engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -39,7 +50,17 @@ def verify_product_activation
             verify_extension_activation(product)
           end
         rescue InstanceVerification::Exception => e
-          raise ActionController::TranslatedError.new('Instance verification failed: %{message}' % { message: e.message })
+          if @system.proxy_byos
+            # BYOS instances that use RMT as a proxy are expected to fail the
+            # instance verification check, however, PAYG instances may send registration
+            # code, as such, instance verification engine checks for those BYOS
+            # instances once instance verification has failed
+            raise ActionController::TranslatedError.new('No subscription with this Registration Code found') unless scc_check_regcode(product)
```

Comment:
Why do we need to do the `scc_check_regcode` check here at all? 
If the regcode is invalid, or doesn't include the requested product then `SccProxy.scc_activate_product` will fail accordingly.